open-craft/ansible-dalite
# Move log rotation and database dumps to separate volume (OC-1589)

Adding plays for formatting and mounting a volume; adding compression to log rotation/database dump script.

**JIRA tickets**: Implements OC-1589

**Discussions**: Discussed in JIRA ticket

**Dependencies**: None

**Sandbox URL**: TBD - sandbox is being provisioned.

**Partner information**: 3rd party-hosted open edX instance

**Testing instructions**:
1. Verify that currently, logs are being stored on the boot volume
2. Create an OpenStack instance with an additional volume attached at `/dev/vdc`
3. Provision that instance with this playbook
4. Verify that logs and DB dumps are being stored on the additional volume

**Author notes and concerns**:
1. Ansible has modules to create and attach OpenStack volumes; I'd like to give those a try once I get access to see if they could be included here.
2. If we don't go that route, I'm guessing there's a better way to pass the volume device ID than I did here; would we just skip defining the variable and have it passed in at runtime?
3. We may want to mount the volume (and thus, the dump location) outside the `logs-download` user's home folder. If that's the case, then I'll add a couple plays to symlink those directories to the `logs-download` user's home folder to preserve continuity.

**Reviewers**
